bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/rewriter/gemini/QueryVisitorRewriter.java

```diff
@@ -0,0 +1,91 @@
+package com.google.zetasql.toolkit.antipattern.rewriter.gemini;
+
+import com.google.zetasql.SqlException;
+import com.google.zetasql.toolkit.antipattern.AntiPatternVisitor;
+import com.google.zetasql.toolkit.antipattern.cmd.InputQuery;
+import com.google.zetasql.toolkit.antipattern.rewriter.prompt.PromptYamlReader;
+import com.google.zetasql.toolkit.antipattern.util.AntiPatternHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class QueryVisitorRewriter {
+    private static final Logger logger = LoggerFactory.getLogger(QueryVisitorRewriter.class);
+    private final PromptYamlReader promptYamlReader;
+    private final AntiPatternHelper antiPatternHelper;
+
+    public QueryVisitorRewriter(AntiPatternHelper antiPatternHelper) throws IOException {
+        this.promptYamlReader = new PromptYamlReader();
+        this.antiPatternHelper = antiPatternHelper;
+    }
+
+    public String rewriteSQL(String inputQuery,
+                             AntiPatternVisitor visitor,
+                             Integer TTL) throws Exception {
```

Comment:
nit. I would prefer another name like retries, remainingAttempt ... I'll leave it up to you.
my concern is that I am used to seeing TTL as a terminology used in networking, so the first thing I think of is of API retries, all I am saying is that TLL might be a confusing name since it a commonly used term for other use cases. I would prefer a name more specific to this use case

Reply:
So TTL is exactly that, the ammount of API uses you are going to use, so not exactly the number of retries, if you set the TTL to 1, it will have 0 retries, so I use this term because it doesn't measure retries but the ammount of calls to the LLM it will do, and is used to avoid infinit loops, kinda like TTL in networking, as each call to the LLM for my represent a NODE in a graph that is following, let me finish the arguments in the command parser, and if it's still confusing, lets change it

Reply:
I can change it to llm retries, and always asume that if rewrite_sql is on, it means the TTL is 1